app/src/main/java/edu/wisc/ece/pinpoint/utils/FirebaseDriver.java

```diff
@@ -210,20 +211,29 @@ public void loadPinImage(ImageView imageView, Context context, String pid) {
                 .into(imageView);
     }
 
-    public Task<String> dropPin(Pin newPin) {
+    public Task<String> dropPin(Pin newPin, Long cost) {
         return functions.getHttpsCallable("dropPin").call(newPin.serialize()).continueWith(task -> {
             String pid = (String) task.getResult().getData();
             pins.put(pid, newPin);
             droppedPinIds.add(pid);
+            pinnies -= cost;
             return pid;
         });
     }
     public Task<Long> getPinnies(String uid) {
-        return db
-                .collection("private")
-                .document(uid)
-                .get()
-                .continueWith(task -> (Long) task.getResult().get("currency"));
+            return db
+                    .collection("private")
+                    .document(uid)
+                    .get()
+                    .continueWith(task -> {
+                        Long pinniesResult = (Long) task.getResult().get("currency");
+                        pinnies = pinniesResult;
+                        return pinniesResult;
+                    });
+    }
+
+    public Long getCachedPinnies(String uid){
+        return pinnies;
     }
```

Comment:
If we're only caching the pinnies for one user we should probably remove the UID argument and just use the current user's UID to avoid confusion.